src/runtime/lib/API.ts

```diff
@@ -177,6 +177,37 @@ export abstract class AbstractWebView {
         }
     }
 
+    public showPersistView(...args: any[]): void {
```

Comment:
Any reason we need to have an additional method created ? I think we can just include the persistence options in the same `showView` method right?

Reply:
Instead of creating a new method altogether, that copies most of the `showView` logic, we should have a way of updating `options` to include this new field `retainContextWhenHidden`. We can probably have something of this sort : 

- Create a protected `shouldPersist` boolean variable and set it to `false` by default.
- On `showView` method, before creating the webview panel, check the `shouldPersist` condition and update `options` map with additional attribute if needed.
- Set the variable to true in simulate skill webview constructor.

What do you think?

Reply:
Thanks Nik for the detailed description, this is a better solution and should go with this. I modified it and please check again :)